### ISSUE TITLE
Next.js environment variables not being set on override

### ISSUE THREAD

**Issue by sananbintahir** [aws-amplify/amplify-console#2097]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

yarn

### If applicable, what version of Node.js are you using?

v14.16.1

### Amplify CLI Version

5.1.0

### What operating system are you using?

Windows

### Amplify Categories

Not applicable

### Amplify Commands

publish

### Describe the bug

When configuring environment variables in amplify for a Next.js project, the way to expose an environment variable to the browser (according to the Next documentation) is by including a prefix `NEXT_PUBLIC_`. This works as expected, except when I want to override the environment variable based on the branch.

After overriding the environment variable, however, when I access it with process.env on the front-end, it still holds the default value.

I add the environment variables in the following way,

```        build:
          commands:
            - echo "NEXT_PUBLIC_API=$NEXT_PUBLIC_API" >> .env
            - yarn run build
```

### Expected behavior

The overridden environment variable should be available after the build, but they are not. The variables do, however, show up correctly in the build logs.

### Reproduction steps

1. Make an Amplify Next.js project
2. Add environment variable in the amplify console (prefix with NEXT_PUBLIC_ so we can test it by logging) e.g NEXT_PUBLIC_API
3. Override the environment variable for a different branch (e.g develop)
4. Change the build settings file to get access to the environment variable,
 ```        build:
          commands:
            - echo "NEXT_PUBLIC_API=$NEXT_PUBLIC_API" >> .env
            - yarn run build
```
5. Run a build for the develop branch
6. console.log(process.env.NEXT_PUBLIC_API) returns the default value instead of the overridden one

### Log output

<details>

```
# Put your logs below this line
2021-07-27T08:28:12.284Z [INFO]: # Executing command: echo "NEXT_PUBLIC_API=<Overriden API value>" >> .env
                                 # Executing command: cat .env
2021-07-27T08:28:12.285Z [INFO]: NEXT_PUBLIC_API=<Overriden API value>
2021-07-27T08:28:12.285Z [INFO]: # Executing command: yarn run build
2021-07-27T08:28:12.410Z [INFO]: yarn run v1.22.0
2021-07-27T08:28:12.426Z [WARNING]: warning ../package.json: No license field
2021-07-27T08:28:12.438Z [INFO]: $ next build
2021-07-27T08:28:12.955Z [INFO]: info  - Loaded env from /codebuild/output/src713826621/src/Farbe-app/client/.env

```

</details>

**Comment by sananbintahir** [comment]
> It appears nextjs environment variables can be set a different way in the amplify console. See [#1987 (comment)](https://github.com/aws-amplify/amplify-console/issues/1987#issuecomment-885217940)
> 
> I'll transfer this issue to them, please note if that comment helps.

Yes exactly, but the issue is the process.env.MY_ENV_VAR (according to the example in the comment) does not get overridden when you override it in the console. To fix it, I had to use dotenv. 

At first, I added the NEXT_PUBLIC_API in the environment variables in the amplify console, including an override for one of the branches.

Then, in the build settings, I did
```
          commands:
            - echo "MY_API=$NEXT_PUBLIC_API" >> .env.custom
            - yarn run build
```

and then in `next.config.js`
```
require("dotenv").config({ path: "./.env.custom" });

module.exports = {
  env: {
    API: process.env.MY_API,
  }
}
```

Then I was able to use process.env.API in my front-end code, with the updated overridden value

**Comment by alacret** [comment]
@sananbintahir this solution didn't work for me. It seems that even doing this:

```js
module.exports = {
  env: {
    NEXT_PUBLIC_TEST:1,
  }
}
```
The variables are not seeing in the browser

**Comment by quinnturner** [comment]
I am also having this exact issue that is preventing us from going to production. I use PR previews with `all branches` setting one value and an override for the `main` branch setting the production value. The `main` branch does not get the overwritten value, it gets the `all branches` value.

**Comment by sananbintahir** [comment]
> @sananbintahir this solution didn't work for me. It seems that even doing this:
> 
> ```js
> module.exports = {
>   env: {
>     NEXT_PUBLIC_TEST:1,
>   }
> }
> ```
> 
> The variables are not seeing in the browser

What i wrote worked for me. Try not using the NEXT_PUBLIC_ in the env variable you’re setting in the module.exports. Following the exact steps that I wrote. But this should be fixed. Its a pretty annoying bug.

**Comment by traianturcu** [comment]
Is there any plan to fix this bug?
